acai-ios/OrderDetailViewController.swift

```diff
@@ -0,0 +1,229 @@
+                result + (option.isSelected ? option.price : 0.0)
+            }
+        }
+        addToCartActionTabView.priceLabel.text = (totalPrice * quantity.doubleValue).asPriceString()
+    }
+
+}
+
+extension OrderDetailViewController: ListAdapterDataSource {
+
+    func objects(for listAdapter: ListAdapter) -> [ListDiffable] {
+
+        var sections: [ListDiffable] = []
+
+        sections.append(EmptyItem(height: FileConstants.emptyItemHeight))
+        
+        for section in menuItem.defaulOptions.keys.sorted(by: { $0 < $1 }) {
+            let options = optionSectionsMap[section] ?? []
+            sections.append(OrderOptions(DiffableArray(options), type: section))
+        }
+
+        sections.append(quantity)
+
+        return sections
+    }
+    
+    func listAdapter(_ listAdapter: ListAdapter, sectionControllerFor object: Any) -> ListSectionController {
+        if let object = object as? EmptyItem {
+            return HeaderImageSectionController(height: object.height, menuItem: menuItem)
+        } else if let object = object as? OrderOptions {
+            let orderCustomizationListSectionController = OrderOptionListSectionController(options: object)
+            orderCustomizationListSectionController.selectOptionDelegate = self
+            return orderCustomizationListSectionController
+        } else if let object = object as? NSNumber {
+            let quantitiyController = QuantitySectionController(quantity: object, itemType: menuItem.type)
+            quantitiyController.delegate = self
+            return quantitiyController
+        }
+        return ListSectionController()
+    }
+    
+    func emptyView(for listAdapter: ListAdapter) -> UIView? {
+        return nil
+    }
+    
+}
+
+extension OrderDetailViewController: DidSelectOptionDelegate {
+    
+    func deselectOption(at index: Int, for type: OrderOption.OptionType) {
+
+        var options = optionSectionsMap[type]!
+        options[index] = options[index].copy(isSelected: false)
+        optionSectionsMap[type] = options
+
+        updateAddToCartPrice()
+        listAdapter.performUpdates(animated: false, completion: nil)
+    }
+    
+    func selectOption(at index: Int, for type: OrderOption.OptionType) {
+
+        var options = optionSectionsMap[type]!
+        let option = options[index]
+        options[index] = option.copy(isSelected: !option.isSelected)
+        optionSectionsMap[type] = options
+
+        updateAddToCartPrice()
+        listAdapter.performUpdates(animated: false, completion: nil)
+
+    }
+    
+}
+
+extension OrderDetailViewController: QuantitySelectionCollectionViewCellDelegate {
+
+    func valueIncremented() {
+        quantity = NSNumber(value: quantity.intValue + 1)
+        updateAddToCartPrice()
+        listAdapter.performUpdates(animated: false, completion: nil)
+    }
+
+    func valueDecremented() {
+        if quantity.intValue > 1 {
+            quantity = NSNumber(value: quantity.intValue - 1)
```

Comment:
why are are we making this `NSNumber`?

Reply:
To "copy" the NSNumber class, we can't modify the same class reference